src/Head2Head.tsx

```diff
@@ -0,0 +1,122 @@
+import { useState } from "react";
+import {
+	getRaces,
+	getSlugData,
+	numPagesForGame,
+	RACETIME,
+	SlugData,
+} from "./Loader";
+import FormattedTable from "./FmtTable";
+import CategoryLauncher from "./CategoryLauncher";
+import "./Head2Head.css";
+import MUTable, { MUEntrant, MUMapCtor } from "./MUTable";
+
+// Used to cancel the async function that loads races if a new slug is set.
+let token: Symbol;
+
+function Head2Head() {
+	const [muTable, setMuTable] = useState(new MUTable());
+	const [playerNameList, setPlayerNameList] = useState<Array<string>>([]);
+	const [recordTable, setRecordTable] = useState(MUMapCtor());
+	const [userInfoMap, setUserInfoMap] = useState(new Map<string, MUEntrant>());
+	const [loading, setLoading] = useState(false);
+	const [validSlug, setValidSlug] = useState(false);
+	const [currentSlugData, setCurrentSlugData] = useState<SlugData>();
+	const [loadingProgress, setLoadingProgress] = useState(0);
```

Comment:
This is a relatively large amount of interconnected / dependent state. This could be cleaner by moving much of this state into a [reducer](https://react.dev/reference/react/useReducer), which is effectively a state machine. This reducer might manage less state, with other info computed as [derived data](https://react.dev/learn/thinking-in-react#step-3-find-the-minimal-but-complete-representation-of-ui-state) - e.g. `recordTable`, `playerNameList` and `userInfoMap` are all derived data in my opinion and thus should not be in a useState or useReducer hook - try to compute them with useMemo.